pom.xml

```diff
@@ -572,5 +572,97 @@
         <skipITs>false</skipITs>
       </properties>
     </profile>
+    <profile>
+      <id>devsite-apidocs</id>
+      <activation>
+        <property>
+          <name>devsite.template</name>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <!-- Generate API docs using Doclava for the developer site. -->
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <!-- Note: version 3.x.x uses additionalOption instead of additionalparam. -->
+            <version>3.1.1</version>
+            <executions>
+              <execution>
+                <phase>site</phase>
+                <goals>
+                  <goal>aggregate</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <docletArtifact>
+                <groupId>com.google.doclava</groupId>
+                <artifactId>doclava</artifactId>
+                <version>1.0.6</version>
+              </docletArtifact>
+              <doclet>com.google.doclava.Doclava</doclet>
+              <bootclasspath>${sun.boot.class.path}</bootclasspath>
+              <additionalDependencies>
+                <additionalDependency>
+                  <groupId>com.google.j2objc</groupId>
+                  <artifactId>j2objc-annotations</artifactId>
+                  <version>1.3</version>
+                </additionalDependency>
+              </additionalDependencies>
+              <additionalOptions>
+                <additionalOption>-hdf</additionalOption>
+                <additionalOption>book.path</additionalOption>
+                <additionalOption>/java/_book.yaml</additionalOption>
+                <additionalOption>-hdf</additionalOption>
+                <additionalOption>project.path</additionalOption>
+                <additionalOption>/_project.yaml</additionalOption>
+                <additionalOption>-hdf</additionalOption>
+                <additionalOption>devsite.path</additionalOption>
+                <additionalOption>/java/docs/reference/</additionalOption>
+                <additionalOption>-d</additionalOption>
+                <additionalOption>${project.build.directory}/apidocs</additionalOption>
```

Comment:
Since this is configuring the javadoc plugin to generate docs using a non-default doclet does it make sense to output to a different directory so that it doesn't collide with the existing javadoc and site generation if they both run in the same build?

Multiple profiles can be active at the same time and I don't think we want this to clobber anything that is already dependent upon someplace else (that we don't know of, or don't have control of).